Networking/Networking/Mapper/JWTokenMapper.swift

```diff
@@ -0,0 +1,90 @@
+
+import Foundation
+
+/// Mapper to parse the JWToken
+///
+struct JWTokenMapper: Mapper {
+    func map(response: Data) throws -> JWToken {
+        let decoder = JSONDecoder()
+        let jwt = try decoder.decode(JWTokenResponse.self, from: response).token
+
+        let payload: [String: Any]? = {
+            let parts = jwt.components(separatedBy: ".")
+            guard parts.count == 3 else {
+                return nil
+            }
+
+            let midPart = parts[1]
+
+            guard let bodyData = Data(base64URLEncoded: midPart) else {
+                return nil
+            }
+
+            guard let json = try? JSONSerialization.jsonObject(with: bodyData, options: []),
+                  let payload = json as? [String: Any] else {
+                return nil
+            }
+            return payload
+        }()
+
+        let expiryDate = try {
+            guard let payload,
+                  let expiry = payload[PayloadKey.expiryDate] as? Double else {
+                throw JWTokenMapperError.expiryDateNotFound
+            }
+
+            return Date(timeIntervalSince1970: expiry)
+        }()
+
+        let siteID =  try {
+            guard let payload,
+                  let siteID = payload[PayloadKey.blodID] as? Int64 else {
+                throw JWTokenMapperError.blogIDNotFound
+            }
+
+            return siteID
+        }()
+
+        return JWToken(token: jwt, expiryDate: expiryDate, siteID: siteID)
+    }
+
+    struct JWTokenResponse: Decodable {
+        let token: String
+    }
+}
+
+private extension JWTokenMapper {
+    enum PayloadKey {
+        static let expiryDate = "exp"
+        static let blodID = "blog_id"
```

Comment:
Typo:
```suggestion
        static let blogID = "blog_id"
```

Reply:
Good catch. Fixed in https://github.com/woocommerce/woocommerce-ios/pull/10896/commits/a873796b43de1ad27811c4eb8c238bb017de1302